### ISSUE TITLE
Text component - improve docs around suggested use

### ISSUE THREAD

**Issue by mcoker** [patternfly/patternfly-react#8413]
TL;DR - we should probably have a blurb that explains what the component is, and what should go in it. Here's something simple - "This component provides simple, built-in styling for putting common blocks of HTML elements together. <TextContent> establishes the block of content, and within it, only the <Text> (and all elements supported by the component prop), <TextList>, and <TextListItem> components can be used to write formatted content. Any other components are not supported as children of <TextContent>."

---

There is confusion around when to use individual components versus their `<Text>` component equivalent/support. A `<List>` vs `<TextList>` or `<Title>` vs `<Text component="h2">`, for example.

The text component (aka [content component](https://github.com/patternfly/patternfly/blob/main/src/patternfly/components/Content/content.scss) in core) essentially lets you do this (core HTML and react component equivalent as a comment) and it will automatically size, format, and space these elements apart. 

```
<div class="pf-c-content"> // `<TextContent>`
  <h1>hello world</h1> // `<Text component="h1">`
  <p>foo paragraph</p> // `<Text component="p">`
  <ul> // `<TextList>`
    <li>apple</li> // `<TextListItem>`
    <li>banana</li>
      <ol> // `<TextList component={TextListVariants.ol}>`
        <li>green</li>  // `<TextListItem>`
        <li>yellow</li>
        <li>spotty</li>
      </ol>
    </li>
  </ul>
</div>
```

Refer to the content component css linked above to see that the styling for text/content elements is based on tag name, not class. So technically anything with those tag names will be styled according to how the text/content component thinks they should be styled. This is for simple content blocks, or cases where you have a heading + paragraph, or multiple paragraphs and/or lists and things, and you want consistent spacing. It is not for placing things like alerts and/or data-lists and other components (probably most components) in, since (for example) a data-list renders as a `<ul>` - you wouldn't want the text/content component's list styling to apply to a data-list.

`<List>` and `<Title>` are for more complex uses - they are individual components and can be used anywhere, and have their own custom properties, variations/features, etc. Technically you _may_ be able to use a simple `<Title>` or `<List>` in a text/content component, but it's probably not a great practice as there may be conflicting styles.

We should probably just make sure the docs for the text component are clear as to what all elements can be used in it. And provide some guidance on when to use stuff from the text component versus individual components that are also in the text component. `<List>`, `<Title>`, `<DescriptionList>`, `<Divider>`... I think that would cover it.

FWIW, here's a basic rundown of the current styling for text component things:

* `p, dl, ol, ul, blockquote, small, pre, table, hr` all get a margin bottom if they aren't the last thing in the content block (there is a margin between them and whatever follows them)
* `h1-h6` get font-size, font-family, line-height, and margin set appropriately based on the heading element used.
* `blockquote` and `small` get some fancy formatting
* `hr` looks like a divider
* `ol, ul, dl` have basic, default list styling
* `a` gets basic link styling, and supports `:visited` styles - both using `:visited` (the browser determines if it has been visited), and with `.pf-m-visited` both on an `a` or on a block containing `a`'s

**Comment by rhamilto** [comment]
Further, if you put a `<List isPlain>` inside a `<TextContent>`, the `isPlain` styling is overridden by the `<TextContent>` styling (margin-left and padding-left).

<img width="1194" alt="Screenshot 2022-12-08 at 10 55 02 AM" src="https://user-images.githubusercontent.com/895728/206497136-edc6c162-a437-443e-b98c-db48dc588271.png">

If you put a `<SimpleList>` inside a `<TextContent>`, the styling is overridden.

<img width="989" alt="Screenshot 2022-12-08 at 10 55 56 AM" src="https://user-images.githubusercontent.com/895728/206497460-2ab0ad3f-944a-4aa8-b1e6-cbc526775801.png">

And `<TextList>` does not have a plain variant.  I will open a bug for this issue.
